### PR TITLE
Feat 2 create icon component

### LINKED ISSUE #4: Create Icon component
# Icon
Create Icon component and create story (with storybook).

## What include ?
- [x] Interface :
    - [x] Icon
    - [x] Paypal 

---

- [x] Enum :
    - [x] Icon
    - [x] Font Size
---

- [x]  Type :
    - [x] Icon
    - [x] Color
    - [x] Font Size

---

- [x] Helper:
    - [x] setIcon Fucntion
    - [x] setPaypalSize Function

### PR DESCRIPTION
**Pull request by eidan66**
# Icon
Create icon component (Atom) & story (storybook) files.

## Files include 🗄️ 📁
* `src\components\atoms\icon\icon.tsx` - The atom. Return the UI.
    * The component receives the following props :
        * icon - return icon element from `material-ui`.

* `src\models\icon\icon.tsx` - Interface file. Init the props of Icon component get.

* `src\types\icon\icon.tsx` - Pick the props from the interface .

* `src\containers\enums\icon.tsx` - Init the icon's element (from the `material ui`) options with enum.
    *  Icon's 🎈 (More icon's will be available on the few next commit's).
        *  Home ('home') - Will change in the next commit's. Home is just for test.

* `src\containers\helpers\icon\icon.tsx`- Include setIcon helper functions.

## What next ?
Before close this PR and marge it to main, there is few more things that i want to add & change.
- [x] Add more icon's :
    - [x] Canceled
    - [x] Paypal - Use with [Font Awesome](https://fontawesome.com/icons/paypal)
    - [x] Income
    - [x] Dashboard
    - [x] Charts
    - [x] Cards
    - [x] Calendar
    - [x] Messages
    - [x] User Profile
    - [x] Settings

- [x] Add color options.
- [x] Add size's options.
- [x] Add stories for each icon.

### CODE REVIEW

**Comment by eidan66** [comment]
## 🔗  [Commit Update](https://github.com/eidan66/Budget-Application/pull/5/commits/a5321346b0c04a63dac808c4322974a6b809e1a3)
In the last commit's i added few things :

* Icon's - Added  more icons to the component:
    * Canceled
    * Paypal
    * Income
    * Dashboard
    * Charts
    * Cards
    * Calendar
    * Messages
    * User Profile
    * Settings

* Color's - Added option to add color. Compatible with :
    * Hex
    * rgb/a
    * string html color name

* Size's - Added size option as Material UI size (value in css font-size ):
    * Default - 1.5rem
    * Inherit - inherit
    * Large - 2.1875rem
    * Small - 1.25rem

* Storybook - Added story for each icon.

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,14 @@
 import React from 'react';
+import * as A from './components/atoms';
```
Comment (almogbhl) [inline src/App.tsx]: import them {}

File: src/App.tsx [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+import * as E from './containers/enums/icon';
 
 const App = () => {
-  return <div>Budget Application</div>;
+  return (
+    <>
+      <A.AtomIcon icon={E.icon.DASHBOARD} color="red" />
```
Comment (almogbhl) [inline src/App.tsx]: import it without A. 
and without the 'atom'
<Icon .../>

File: src/App.tsx [hunk 3]
```diff
@@ -1,7 +1,14 @@
 import React from 'react';
+import * as A from './components/atoms';
+import * as E from './containers/enums/icon';
```
Comment (almogbhl) [inline src/App.tsx]: *idiomatic code*

please follow the best practices for this -  find and share e with us how to import enums - the convention

File: src/components/atoms/AtomIcon/AtomIcon.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import { Story, Meta } from '@storybook/react/types-6-0';
+import AtomIcon from './';
+import * as I from '../../../models/icon/icon';
+import * as E from '../../../containers/enums/icon';
+export default {
+  title: 'Typography',
+  component: AtomIcon,
+} as Meta;
+
+const Template: Story<I.Icon> = (args) => <AtomIcon {...args} />;
```
Comment (almogbhl) [inline src/components/atoms/AtomIcon/AtomIcon.stories.tsx]: please check why we use the ```template.bind```

what's the point of it and why not just use:

```

export default {
  title: 'Components/atoms/Icon',
};

export const Canceled = () => <Icon ... />;

```

**Pull request merged**